crates/sui/src/client_commands.rs

```diff
@@ -3098,3 +3107,66 @@ async fn check_protocol_version_and_warn(client: &SuiClient) -> Result<(), anyho
 
     Ok(())
 }
+
+/// Fetch move packages based on the provided package IDs.
+async fn fetch_move_packages(
+    client: &SuiClient,
+    package_ids: &[ObjectID],
+) -> Result<Vec<MovePackage>, anyhow::Error> {
+    let objects = client
+        .read_api()
+        .multi_get_object_with_options(package_ids.to_vec(), SuiObjectDataOptions::bcs_lossless())
+        .await
+        .map_err(|e| anyhow!("{e}"))?;
+
+    objects
+        .into_iter()
+        .map(|o| {
+            let o = o.into_object().map_err(|e| anyhow!("{e}"))?;
+            let Some(SuiRawData::Package(p)) = o.bcs else {
+                bail!("Expected SuiRawData::Package but got something else");
```

Comment:
This is a user facing error, they may hit it if a dependency mistakenly indicated that they were published at a Move Object instead of a Package. As such, you should:

- Not mention internal details, like `SuiRawData` -- phrase the issue in terms of user-facing terms ("Object is not a package").
- Give enough context for them to track down the issue -- they need to know which package ID failed to fetch, and ideally they need to know which package name that corresponds to as well.